pandas/_libs/tslibs/conversion.pxd

```diff
@@ -33,3 +32,21 @@ cdef maybe_datetimelike_to_i8(object val)
 cdef int64_t tz_convert_utc_to_tzlocal(int64_t utc_val, tzinfo tz)
 
 cpdef datetime localize_pydatetime(datetime dt, object tz)
+
+cdef extern from "numpy/ndarraytypes.h":
```

Comment:
Since this is done in np_datetime.pxd, I'd just cimport from there rather than copying this.